File: data/src/main/java/com/radioteria/data/entities/File.java
package com.radioteria.data.entities;

import javax.persistence.*;

@Entity
@Table(name = "files")
public class File extends BaseEntity<Long> {
    final public static String ID = "id";
    final public static String NAME = "name";
    final public static String CONTENT_ID = "content_id";

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = ID, nullable = false, updatable = false)
    private Long id;

    @Column(name = NAME, nullable = false)
    private String name;

    @ManyToOne(targetEntity = Content.class)
    @JoinColumn(name = CONTENT_ID, nullable = false)
    private Content content;

    @Override
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Content getContent() {
        return content;
    }

    public void setContent(Content content) {
        this.content = content;
    }
}

File: db/src/main/kotlin/com/radioteria/db/entities/File.kt
package com.radioteria.db.entities

import javax.persistence.*

object FileMeta {
    const val TABLE_NAME = "files"
    const val NAME = "name"
    const val CONTENT_ID = "content_id"
}

@Entity
@Table(name = FileMeta.TABLE_NAME)
@Access(AccessType.FIELD)
class File(
        @ManyToOne(targetEntity = Content::class)
        @JoinColumn(name = FileMeta.CONTENT_ID, nullable = false)
        var content: Content,

        @Column(name = FileMeta.NAME, nullable = false)
        var name: String = "",

        id: Long? = null
) : IdAwareEntity<Long>(id)